pystac/extensions/projection.py

```diff
@@ -96,7 +104,15 @@ def apply(
             transform : The affine transformation coefficients for
                 the default grid
         """
-        self.epsg = epsg
+        if epsg is not None and code is not None:
+            raise KeyError(
+                "Only one of the options ``code`` and ``epsg`` should be specified."
+            )
+        elif epsg:
+            self.epsg = epsg
```

Comment:
🤔 down in the [setter](https://github.com/stac-utils/pystac/pull/1287/files#diff-5718f1829c91c51869d65fe57f84b3405efb887f0db416197ca48395576dd022R142) we set `code`, but here we set `epsg`. Is there a reason those are different?

Reply:
This is using the `epsg` setter. It's basically saving having to write `self.code = f"EPSG:{epsg}"` in two places.